File: MarxUp/MarxUp/ToolboxInitializer.m
//
//  ToolboxInitializer.m
//  MarxUp
//
//  Created by Ognyanka Boneva on 27.09.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

#import "ToolboxInitializer.h"
#import "ToolboxDataSource.h"


@interface ToolboxInitializer ()

@property (strong, nonatomic) ToolboxDataSource *data;

@end


@implementation ToolboxInitializer

+ (instancetype)newToolboxInitializer {
    ToolboxInitializer *new = [ToolboxInitializer new];
    if (new) {
        new.data = [ToolboxDataSource newDataSource];
    }
    return new;
}

- (void)addToolboxItemsToView:(UIScrollView *)toolboxView withTarget:(UIViewController *)targetController andSelector:(SEL)selector {
    int buttonY = 0;
    CGFloat buttonSize = toolboxView.frame.size.width;
    
    for (int i = 0; i < [self.data numberOfToolboxItems]; i++) {
        UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, buttonY, buttonSize, buttonSize)];
        [self.data initItemWithButton:button atIndex:i];
        
        [button addTarget:targetController action:selector forControlEvents:UIControlEventTouchUpInside];
        [toolboxView addSubview:button];
        
        buttonY = buttonY + buttonSize;
    }
    
    toolboxView.contentSize = CGSizeMake(toolboxView.frame.size.width, buttonY);
    [ToolboxInitializer configUI:toolboxView];
}

+ (void)configUI:(UIScrollView *)view {
    view.backgroundColor = [UIColor lightGrayColor];
    view.layer.borderColor = [UIColor orangeColor].CGColor;
    view.layer.shadowColor = [UIColor lightGrayColor].CGColor;
    view.layer.shadowRadius = 1.0;
    view.layer.shadowOpacity = 0.5;
    view.layer.borderWidth = 2.0;
}

@end

File: MarxUpUp/MarxUpUp/ToolboxInitializer.swift
//
//  ToolboxInitializer.swift
//  MarxUpUp
//
//  Created by Ognyanka Boneva on 20.11.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

import UIKit

class ToolboxInitializer: NSObject {

    static var buttonSize: Int {
        return 50
    }
    
    static func addToolboxItems(toView scrollView: UIStackView?, withTarget targetController: ToolboxStackController, _ selector: Selector, andDataSource dataSource: ToolboxDataSource) {
        guard let view = scrollView else {
            print("Error: Source view is nil")
            return
        }
        
        let _ = Array(0..<dataSource.itemsCount).map {
            let button = UIButton(frame: CGRect(x: 0, y: 0, width: buttonSize, height: buttonSize))
            
            button.tag = $0
            button.setImage(dataSource.item(atIndex: $0).image, for: UIControl.State.normal)
            button.addTarget(targetController, action: selector, for: UIControl.Event.touchDown)
            
            view.addArrangedSubview(button)
        }
        
        view.superview?.layer.cornerRadius = 10
        view.backgroundColor = UIColor.black.withAlphaComponent(0.7)
    }
}
